src/main/kotlin/com/openlattice/directory/UserDirectoryService.kt

```diff
@@ -22,40 +22,78 @@
 package com.openlattice.directory
 
 import com.auth0.json.mgmt.users.User
+import com.codahale.metrics.annotation.Timed
 import com.google.common.collect.ImmutableMap
 import com.hazelcast.core.HazelcastInstance
 import com.hazelcast.core.IMap
 import com.openlattice.auth0.Auth0TokenProvider
+import com.openlattice.authentication.Auth0Configuration
 import com.openlattice.client.RetrofitFactory
 import com.openlattice.datastore.services.Auth0ManagementApi
 import com.openlattice.directory.pojo.Auth0UserBasic
 import com.openlattice.hazelcast.HazelcastMap
 import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
 
 
-open class UserDirectoryService(auth0TokenProvider: Auth0TokenProvider, hazelcastInstance: HazelcastInstance) {
-    companion object {
-        private val logger = LoggerFactory.getLogger(UserDirectoryService::class.java)
-        const val DEFAULT_PAGE_SIZE = 100
+interface UserDirectoryService {
+    @Timed
+    fun getAllUsers(): Map<String, User>
+
+    @Timed
+    fun getUser(userId: String): User
+
+    //TODO: Switch over to a Hazelcast map to relieve pressure from Auth0
+    @Timed
+    fun searchAllUsers(searchQuery: String): Map<String, Auth0UserBasic>
```

Comment:
I think we could switch over and use User here as well. If I remember right, it has the "missing" properties named `roles` and `com/openlattice/organization` (also, is this intentional or a typo? 🙃  ) in `app_metadata`

Reply:
https://jira.openlattice.com/browse/LATTICE-2361

Reply:
Oh, just noticed, that the non-local one still has the old implementation...nvm then